### ISSUE TITLE
shadowJar task with minimize() doesn't use gradle toolchain.

### ISSUE THREAD

**Issue by VasilisManol** [GradleUp/shadow#1034]
### Expected and Results

call shadowJar task with minimize() without errors.

### Related environent and versions

Gradle 8.6
shadow 8.3.5

Installed JDKs
openjdk 8 - default
openjdk 21

### Reproduction steps

- Use jdk 8 as default
- Install jdk 21 
- Create a gradle project with build.gradle.kts:
```kts
plugins {
    id("java")
    id("com.gradleup.shadow") version ("8.3.5")
}

java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(21))
    }
}

configure<JavaPluginExtension> {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_21
}

repositories {
    mavenCentral()
    mavenLocal()
}

dependencies {
....
}

tasks.named<com.github.jengelman.gradle.plugins.shadow.tasks.ShadowJar>("shadowJar") {
    exclude("META-INF", "META-INF/**")
    minimize()
    archiveClassifier.set("")
    isZip64 = true
}

tasks {
    build {
        dependsOn(shadowJar)
    }
}
```

- Build e.g. ./gradlew build
  Then the following error occurs:
```
  Execution failed for task ':shadowJar'.
> org/vafer/jdependency/Clazzpath has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```
If minimize() is commented out it builds as expected. 

### Anything else?

This error looks related with https://github.com/GradleUp/shadow/issues/814 but maybe not exactly the same. When switching to jdk21 as default jdk, minimize() doesn't give any error.

**Comment by Goooler** [comment]
@VasilisManol This looks unrelated to Shadow, can you try out with deleting

```kt
configure<JavaPluginExtension> {
    sourceCompatibility = JavaVersion.VERSION_1_8
    targetCompatibility = JavaVersion.VERSION_21
}
```

**Comment by VasilisManol** [comment]
@Goooler I get the same even when I comment out that part. Even in that case, it builds without errors, if I just comment out the minimize(). Can you explain why you think it is unrelated to Shadow? For me, the fact that it fails only if I call minimize() but it works otherwise, points to Shadow and actually the implementation of the minimize() feature.

**Comment by VasilisManol** [comment]
I see your point but then it should work with your suggestion. I think the issue is that Shadow doesn't use the toolchain defined in```
 java {
    toolchain {
        languageVersion.set(JavaLanguageVersion.of(21))
    }
}
```
As I wrote my default java on the system is java 8 and this is what Shadow uses. If I set default java to 21 then it works.

**Comment by Goooler** [comment]
`ShadowJar` extends `org.gradle.jvm.tasks.Jar`, I don't see how to configure and use toolchain in that, need more time to investigate...

**Comment by Goooler** [comment]
Any reproducer for this?

**Comment by Goooler** [comment]
Feel free to reopen this if you can repro it and provide a reproducer.

**Comment by VasilisManol** [comment]
Hi, @Goooler sorry I missed your previous comment. I can still reproduce it using the steps that I left in the description. Have you tried that? I will try to make it clearer. I am using Ubuntu 24.04 so assume that for the following commands. 
- Install 2 java toolchains  and use openjdk-8 as default:
```bash
sudo apt install openjdk-8-jdk openjdk-21-jdk
sudo update-alternatives --config 'java' 
# select java 8
```
This way java 8 is the default on the terminal ( can be verified with java --version )
- Create a minimal gradle project. I used gradle 8.13 and ./gradlew init with defaults ( java, kotlin, etc ). 
- Use the build.gradle.kts with the one in the issue description, keep the dependencies that were before.
- Build with ./gradlew build and you will get the error. 
- Comment the minimize() line and the project will be built ok.

I have tested as you suggested with 
sourceCompatibility = JavaVersion.VERSION_11  and 21 and without any but still the same error.

Can you reproduce this?

**Comment by Goooler** [comment]
I believe it's due to the requirement from https://github.com/tcurdt/jdependency/blob/master/HISTORY.md#version-2110-release-18112024.

I bumped the min Java level to 11 in #1242, please try out Java 11 or above.

**Comment by VasilisManol** [comment]
Yes, it works for 11. Thanks for the help! Soon, we will move out completely from java 8 so no worries :)
